test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java

```diff
@@ -159,6 +159,8 @@ protected RepositoryData getRepositoryData(String repoName, Version version) {
         final RepositoryData repositoryData = getRepositoryData(repoName);
         if (SnapshotsService.includesRepositoryUuid(version) == false) {
             return repositoryData.withoutUuid();
```

Comment:
This seems a bit trappy, let's rename this and have it just remove the one UUID:

```suggestion
            return repositoryData.withoutRepositoryUuid().withoutClusterUuid();
```

We can sort it out after the backport so that there's just one method that removes both UUIDs.

Reply:
Jup added a TODO for that, we can fold a few things together after the backport I think :)